packages/react-instantsearch-hooks/src/lib/useSearchState.ts

```diff
@@ -0,0 +1,74 @@
+import { useConnector } from '../hooks/useConnector';
+
+import type { Connector, IndexUiState, UiState } from 'instantsearch.js';
+
+type SearchStateRenderState = {
+  uiState: UiState;
+  indexUiState: IndexUiState;
+  setUiState(uiState: UiState | ((previousUiState: UiState) => UiState)): void;
+  setIndexUiState(
+    indexUiState:
+      | IndexUiState
+      | ((previousIndexUiState: IndexUiState) => IndexUiState)
+  ): void;
+};
+
+type SearchStateWidgetDescription = {
+  $$type: 'ais.searchState';
+  renderState: SearchStateRenderState;
+};
+
+const connectSearchState: Connector<SearchStateWidgetDescription, never> = (
+  renderFn
+) => {
+  return (widgetParams) => {
+    let setUiState: SearchStateRenderState['setUiState'];
+    let setIndexUiState: SearchStateRenderState['setIndexUiState'];
+    return {
+      $$type: 'ais.searchState',
+      getWidgetRenderState({ instantSearchInstance, parent }) {
+        const indexId = parent!.getIndexId();
+        const uiState = instantSearchInstance.getUiState();
+        const indexUiState = uiState[indexId];
+        if (!setUiState) {
+          setUiState = instantSearchInstance.setUiState.bind(
+            instantSearchInstance
+          );
+        }
+        if (!setIndexUiState) {
+          setIndexUiState = (nextUiState) => {
+            setUiState((prevUiState) => ({
+              ...prevUiState,
+              [indexId]:
+                typeof nextUiState === 'function'
+                  ? nextUiState(prevUiState[indexId])
+                  : nextUiState,
+            }));
+          };
+        }
```

Comment:
Are there advantages in defining those setters in the `init()` method of the connector, instead of doing this conditionally here?

Reply:
init gets called after getWidgetRenderState in some cases :)